lib/luadata.c

```diff
@@ -50,41 +51,76 @@ static inline luadata_t *luadata_checkdata(lua_State *L, lua_Integer *offset, lu
 	return data;
 }
 
-static int luadata_getnumber(lua_State *L)
-{
-	lua_Integer offset;
-	luadata_t *data = luadata_checkdata(L, &offset, LUADATA_NUMBER_SZ);
+#define LUADATA_NEWGETTER(what) 	\
+static int luadata_get##what(lua_State *L) \
+{					\
+	lua_Integer offset;		\
+	luadata_t *data = luadata_checkdata(L, &offset, sizeof(what##_t));	\
+					\
+	lua_pushinteger(L, (lua_Integer)*(what##_t *)(data->ptr + offset));	\
+	return 1;			\
+}
 
-	lua_pushinteger(L, *(lua_Integer *)(data->ptr + offset));
-	return 1;
+LUADATA_NEWGETTER(int8);
+LUADATA_NEWGETTER(uint8);
+LUADATA_NEWGETTER(int16);
+LUADATA_NEWGETTER(uint16);
+LUADATA_NEWGETTER(int32);
+LUADATA_NEWGETTER(uint32);
+
+#define LUADATA_NEWSETTER(what)			\
+static int luadata_set##what(lua_State *L)	\
+{						\
+	lua_Integer offset;			\
+	luadata_t *data = luadata_checkdata(L, &offset, sizeof(what##_t));	\
+						\
+	*(what##_t *)(data->ptr + offset) = (what##_t)luaL_checkinteger(L, 3);	\
+	return 0;				\
 }
 
-static int luadata_setnumber(lua_State *L)
-{
-	lua_Integer offset;
-	luadata_t *data = luadata_checkdata(L, &offset, LUADATA_NUMBER_SZ);
+LUADATA_NEWSETTER(int8);
+LUADATA_NEWSETTER(uint8);
+LUADATA_NEWSETTER(int16);
+LUADATA_NEWSETTER(uint16);
+LUADATA_NEWSETTER(int32);
+LUADATA_NEWSETTER(uint32);
+
+#ifdef __LP64__
+LUADATA_NEWGETTER(int64);
+LUADATA_NEWGETTER(uint64);
+LUADATA_NEWSETTER(int64);
+LUADATA_NEWSETTER(uint64);
+#endif
+
+static inline int luadata_getbyte(lua_State *L) {
+	return luadata_getuint8(L);
+}
 
-	*(lua_Integer *)(data->ptr + offset) = luaL_checkinteger(L, 3);
-	return 0;
+static inline int luadata_setbyte(lua_State *L) {
+	return luadata_setuint8(L);
 }
 
-static int luadata_getbyte(lua_State *L)
+#ifdef __LP64__
+static inline int luadata_getnumber(lua_State *L)
```

Comment:
I think we can deprecate it.. as it can be a simply alias in lua (e.g., `local getnumber = data.getint64`).. what do you think?